pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -63,21 +96,52 @@ func (s *Service) Handle(msg dispatcher.DIDCommMsg) error {
 	if !current.CanTransitionTo(next) {
 		return fmt.Errorf("invalid state transition: %s -> %s", current.Name(), next.Name())
 	}
-	// TODO: call pre-transition listeners -  Issue: https://github.com/hyperledger/aries-framework-go/issues/140
-	followup, err := next.Execute(msg)
+
+	// trigger the event if the message type is registered (if it's not outbound)
+	eventsTriggered := false
+	if !msg.Outbound {
+		eventsTriggered, err = s.sendEvent(msg, thid, next)
```

Comment:
I assume eventsTriggered is only set to true for events that have callback actions (e.g., not error events).

Reply:
At the moment, events are triggered only for connectionRequest, response and ack. [Ref](https://github.com/hyperledger/aries-framework-go/pull/240/files#diff-f761c91e753eba1c42a68076df06ce12R162)